Model/Product/Search/Request/Container/Filter/CustomerGroup.php

```diff
@@ -0,0 +1,127 @@
+use Smile\ElasticsuiteCore\Search\Request\QueryInterface;
+
+/**
+ * Product Customer Group Default filter.
+ *
+ * @category Smile
+ * @package  Smile\ElasticsuiteSharedCatalog
+ * @author   Pierre Gauthier <pierre.gauthier@smile.fr>
+ */
+class CustomerGroup implements FilterInterface
+{
+    /**
+     * @var QueryFactory
+     */
+    private $queryFactory;
+
+    /**
+     * @var CompanyContext
+     */
+    private $companyContext;
+
+    /**
+     * @var Config
+     */
+    private $config;
+
+    /**
+     * @var CustomerGroupManagement
+     */
+    private $customerGroupManagement;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @var boolean[]
+     */
+    private $sharedCatalogStatusByGroup = [];
+
+    /**
+     * Customer group filter constructor.
+     *
+     * @param QueryFactory            $queryFactory            Query Factory
+     * @param CompanyContext          $companyContext          Company context
+     * @param Config                  $config                  Shared catalog config.
+     * @param CustomerGroupManagement $customerGroupManagement Customer group manager.
+     * @param StoreManagerInterface   $storeManager            Store manager.
+     */
+    public function __construct(
+        QueryFactory $queryFactory,
+        CompanyContext $companyContext,
+        Config $config,
+        CustomerGroupManagement $customerGroupManagement,
+        StoreManagerInterface $storeManager
+    ) {
+        $this->queryFactory            = $queryFactory;
+        $this->companyContext          = $companyContext;
+        $this->config                  = $config;
+        $this->customerGroupManagement = $customerGroupManagement;
+        $this->storeManager            = $storeManager;
+    }
+
+    /**
+     * {@inheritdoc}
+     * @throws LocalizedException
+     */
+    public function getFilterQuery()
+    {
+        $customerGroupId = $this->companyContext->getCustomerGroupId();
+        if (!$this->isActive($customerGroupId)) {
+            return null;
+        }
+        return $this->queryFactory->create(
+            QueryInterface::TYPE_TERM,
+            [
+                'field' => 'customer_group_id',
+                'value' => $customerGroupId,
+            ]
+        );
+    }
+
+    /**
+     * Check if the shared catalog feature is active for the customer group id.
+     *
+     * @param int $customerGroupId Customer group id.
+     *
+     * @return boolean
+     * @throws LocalizedException
+     */
+    private function isActive($customerGroupId)
+    {
+        if (!isset($this->sharedCatalogStatusByGroup[$customerGroupId])) {
+            $websiteId                = $this->storeManager->getWebsite()->getId();
+            $isActive                 = $this->config->isActive(ScopeInterface::SCOPE_WEBSITE, $websiteId);
+            $isMasterCatalogAvailable = $this->customerGroupManagement->isMasterCatalogAvailable($customerGroupId);
```

Comment:
Beware, we had a "backward compatibility fix" here : https://github.com/Smile-SA/magento2-module-elasticsuite-shared-catalog/pull/10/files

I think we should keep it, since method has a different name across Magento versions.